watcher/pages/renderer.js

```diff
@@ -0,0 +1,65 @@
+const watchFileInput = () => {
+  const fileInput = document.querySelector('.input-file input')
+  const fileLabel = document.querySelector('.input-file__text')
+
+  fileInput.addEventListener('change', (e) => {
+    // show selected directory
+    const path = e.target.value.split('\\')
+    fileLabel.innerHTML = path[path.length - 1]
+  })
+}
+
+const watchForm = () => {
+  const form = document.querySelector('form')
+  const submitButton = form.querySelector('input[type=submit]')
+  const fileInput = form.querySelector('.input-file input')
+  const fileLabel = form.querySelector('.input-file__text')
+  const result = document.querySelector('.result')
+
+  form.addEventListener('submit', (e) => {
+    e.preventDefault()
+
+    submitButton.disabled = true
+    submitButton.innerHTML = 'Uploading...'
+    result.classList.remove('error')
+
+    const formData = new FormData()
+    formData.append('file', fileInput.files[0])
+
+    fetch('/upload', {
```

Comment:
This is placeholder for actual functionality. I am adding helia so you can ignore this part anything in this function after line 26. Just check the then/catch callbacks.